### ISSUE TITLE
QmlAV.dll is not valid Win32 application

### ISSUE THREAD

**Issue by fhazubski** [wang-bin/QtAV#808]
I have taken QtAV source code (v1.11 tag) and recommended ffmpeg package (ffmpeg-3.2.1-windows-x86+x64). I have followed the build instructions (https://github.com/wang-bin/QtAV/wiki/Build-QtAV). Project was built correctly. I have ran "sdk_install.bat". Then when I tried to run an app that uses QtAV and I got an error "C:\Qt\5.6\msvc2013_64\qml\QtAV\QmlAV.dll is not valid Win32 application" and qml failed to load.

I repeated the steps above for several Qt kits and got following results:
- Qt 5.6.2 MinGW 4.9.2 32-bit Debug - working fine
- Qt 5.6.2 MinGW 4.9.2 32-bit Release - crashing right after starting - needs proper DLLs in build directory to run correctly
- Qt 5.6.2 MSVC2013 32-bit Debug - working fine
- Qt 5.6.2 MSVC2013 32-bit Release - crashing right after starting - needs proper DLLs in build directory to run correctly
- Qt 5.6.2 MSVC2013 64-bit Debug - qml\QtAV\QmlAV.dll is not valid Win32 application
- Qt 5.6.2 MSVC2013 64-bit Release - qml\QtAV\QmlAVd.dll is not valid Win32 application

I am using Windows 10 64-bit
Do you have any ideas of what am I doing wrong? Thanks

**Comment by fhazubski** [comment]
DLL from [this build](https://ci.appveyor.com/project/wang-bin/qtav/build/job/8l749hep88iytcdg/artifacts) is working fine, so the problem is on my side.

If I add 
`LIBS += -L$$PWD/../ffmpeg-3.2.1-windows-x86+x64/lib`
to ".qmake.conf" I get build error. If I add
`LIBS += -L$$PWD/../ffmpeg-3.2.1-windows-x86+x64/lib/x64`
to ".qmake.conf" the project builds but the DLL is not valid Win32 application. What I do with the QtAV project is I just open it with Qt Creator, run qmake, and run rebuild. What am I missing?

**Comment by wang-bin** [comment]
What build error?

**Comment by fhazubski** [comment]
> What build error?

I have no build error. I build QtAV with no errors, then run "sdk_install.bat". Then if I try to run an application that uses QtAV and the application tries to load QmlAV.dll I have an error that this DLL is not valid Win32 application. Something like this in console: "The component (.qml file) could not be loaded because it tries to load QmlAV.dll and QmlAV.dll is not valid Win32 application".

I tested this problem with several Qt kits, as listed in the first post and I have the problem only on 64-bit builds.

**Comment by fhazubski** [comment]
I have attached above the errors I get. If there is anything I can share with you to troubleshoot problem faster please do not hesitate telling it. I have tried to reinstall Qt 5.6 kit and it did not change anything

**Comment by wang-bin** [comment]
You should use `LIBS += -L$$PWD/../ffmpeg-3.2.1-windows-x86+x64/lib/x64`. 32bit libs are ignored by the linker, so you got those errors.

I don't know why your dll is invalid. Only QmlAV.dll? What about QtAV1.dll and QtAVWidgets1.dll?

**Comment by fhazubski** [comment]
I have not tested QtAV1.dll and QtAVWidgets1.dll. I will test them and post results

**Comment by wang-bin** [comment]
Try Player.exe

**Comment by fhazubski** [comment]
I do not want to sound stupid but it works now. For all the Qt kits I listed in the first post I have tested player subproject, QMLPlayer subproject and my own project that uses QmlAV.dll.

The only thing that was not working was my own project and MinGW Release kit. App needed proper DLL files in build directory, but I will find a way around this.

What could solve my problem is that yesterday I have updated my Qt components. I was not installing anything new, just updated already installed. For example updated Qt Creator 4.2.0 to 4.2.1

Sorry for the trouble but thank you a lot for your help!
